src/plugins/telemetry/server/saved_objects/get_telemetry_saved_object.ts

```diff
@@ -6,28 +6,27 @@
  * Side Public License, v 1.
  */
 
-import { SavedObjectsErrorHelpers, type SavedObjectsClientContract } from '@kbn/core/server';
-import type { TelemetrySavedObject } from '.';
+import type { SavedObjectsClientContract } from '@kbn/core-saved-objects-api-server';
+import { SavedObjectsErrorHelpers } from '@kbn/core-saved-objects-utils-server';
+import type { TelemetrySavedObject } from './types';
+import { TELEMETRY_SAVED_OBJECT_TYPE, TELEMETRY_SAVED_OBJECT_ID } from './constants';
 
 type GetTelemetrySavedObject = (
-  repository: SavedObjectsClientContract
+  soClient: SavedObjectsClientContract
 ) => Promise<TelemetrySavedObject>;
 
 export const getTelemetrySavedObject: GetTelemetrySavedObject = async (
-  repository: SavedObjectsClientContract
+  soClient: SavedObjectsClientContract
 ) => {
   try {
-    const { attributes } = await repository.get<TelemetrySavedObject>('telemetry', 'telemetry');
+    const { attributes } = await soClient.get<TelemetrySavedObject>(
+      TELEMETRY_SAVED_OBJECT_TYPE,
+      TELEMETRY_SAVED_OBJECT_ID
+    );
     return attributes;
   } catch (error) {
     if (SavedObjectsErrorHelpers.isNotFoundError(error)) {
-      return null;
```

Comment:
Removing this because we relied on `null` when there was not a SO (for Opt-In handling). But we store so many things now in the SO, that we may return something and still not have a clear opt-in property stored in the SO, so it's better to check the property itself.

I plan to take https://github.com/elastic/kibana/issues/135107 to fix this existing bug.